src/sentry/static/sentry/app/views/organizationEventsV2/utils.jsx

```diff
@@ -17,18 +19,42 @@ export function getCurrentView(requestedView) {
  * @param {Object} view
  * @returns {Object}
  */
-export function getQuery(view) {
-  const data = {...view.data};
-  const fields = data.fields.reduce((list, field) => {
+export function getQuery(view, location) {
+  const fields = [];
+  const aggregations = view.data.aggregations ? [...view.data.aggregations] : [];
+  const groupby = view.data.groupby ? [...view.data.groupby] : [];
+
+  view.data.fields.forEach(field => {
     if (SPECIAL_FIELDS.hasOwnProperty(field)) {
-      list.push(...SPECIAL_FIELDS[field].fields);
+      const specialField = SPECIAL_FIELDS[field];
+
+      if (specialField.hasOwnProperty('fields')) {
+        fields.push(...specialField.fields);
+      }
+      if (specialField.hasOwnProperty('aggregations')) {
+        aggregations.push(...specialField.aggregations);
```

Comment:
Do you need this now that aggregations are not in the client?

Reply:
I implemented a version that treats special fields defined on the client and on the API in the exact same way. We could remove the option to pass aggregations on the client since we don't use it at all though

Reply:
Updated to remove any reference to aggregations on the client. It does look a bit cleaner now - thanks for the feedback